tools/bridge/bridge/confirmation_sender.py

```diff
@@ -59,6 +60,12 @@ def run(self):
                 greenlet.kill()
 
 
+sender_retry = tenacity.retry(
```

Comment:
A little late, but I've noticed there's also a [middleware](https://web3py.readthedocs.io/en/stable/middleware.html#httprequestretry) provided by web3 that retries requests. There doesn't seem to be any configuration options, but at least it would be used automatically if we enable it once. It's also possible to implement [custom middlewares](https://web3py.readthedocs.io/en/stable/internals.html#internals-middlewares).

Reply:
At the moment I see no good reason to switch. The nice thing about tenacity is that it would also work with a redis client (if  @saschagoebel succeeds with sneaking in redis) :)

Reply:
We could have a tenacity retry middleware:

```Py
def tenacity_retry_middleware(make_request, w3):
    make_request_retry = retry(make_request)
    def middleware(method, params):
        return make_request_retry(method, params)
    return middleware

w3.middleware_onion.add(tenacity_retry_middleware)
```

The main advantage would be that it would be used automatically anytime we use the web3 object. So we wouldn't need all those custom RPC request methods.

Reply:
I like to separate functions doing RPC requests from those that don't as much as possible, e.g. b924731c6f12e75a15a954aafd27beefd726e6c2
Having it explicit helps when refactoring such stuff.